src/main/java/org/jgroups/protocols/kubernetes/Client.java

```diff
@@ -92,22 +99,28 @@ protected String fetchFromKubernetes(String op, String namespace, String labels,
             return Collections.emptyList();
         }
         List<Json> items=json.at("items").asJsonList();
-        List<String> pods=new ArrayList<>();
+        List<Pod> pods=new ArrayList<>();
         for(Json obj: items) {
-            if(obj.isObject() && obj.has("status")) {
-                Json status=obj.at("status");
-                if(status.isObject() && status.has("podIP")) {
-                    String podIP=status.at("podIP").asString();
-                    if(status.has("phase")) {
-                        Json phase=status.at("phase");
-                        if(phase != null && phase.isString() && !"Running".equals(phase.asString())) {
-                            log.trace("skipped pod with IP=%s as it is not running (%s)", podIP, phase);
-                            continue;
-                        }
-                    }
-                    if(!pods.contains(podIP))
-                        pods.add(podIP);
-                }
+            String parentDeployment = Optional.ofNullable(obj.at("metadata"))
```

Comment:
Big change: I'd like to test this with minikube first... however the turnaround time for this is quite high .. stay tuned.